### PR TITLE
[#153119909] Upgrade ruby to 2.5

### PR DESCRIPTION
**Pull request by bandesz**
## What

Update Ruby version to 2.5 for the project and for all Docker images.

Ruby 2.2 reached EOL on 2018-03-31 (#1), so we decided to migrate to the latest stable version.

The PR affects the following Docker images:

- bosh-cli-v2
    * change the base image to ruby:2.5-slim
- bosh-shell
- bosh
- cf-cli
    * change the base image to ruby:2.5-alpine
- cf-uaac
- json-minify
- self-update-pipelines

We also had to update the json gem to 1.8.6 as installing 1.8.1 (or 1.8.3 as commented) had errors on OSX with Ruby 2.5.1.

## How to review

Code review and review together with the paas-cf and paas-bootstrap PRs.

## Who can review it

Not me.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,5 +11,4 @@ gem 'should_not', '~> 1.1'
 
 gem 'rake', '~> 10.4.2'
 
-# The default latest json 1.8.3 fails on some macs
-gem 'json', '= 1.8.1'
+gem 'json', '= 1.8.6'
```
Comment (alext) [inline Gemfile]: Should we use 2.1.0 of this as that's what ships with Ruby 2.5.1

Comment (bandesz) [reply Gemfile]: As this is used only in the tests there is not much risk to go for 2.1.0

**Comment by henrytk** [comment]
I am getting an error locally when running `rake spec:cf-uaac`:

```
$ rake spec:cf-uaac
/Users/henryknott/.rvm/rubies/ruby-2.2.1/bin/ruby -I/Users/henryknott/.rvm/gems/ruby-2.2.1/gems/rspec-support-3.5.0/lib:/Users/henryknott/.rvm/gems/ruby-2.2.1/gems/rspec-core-3.5.4/lib /Users/henryknott/.rvm/gems/ruby-2.2.1/gems/rspec-core-3.5.4/exe/rspec --pattern cf-uaac/\*\*/\*_spec.rb

cf-uaac image
  has the expected version of uaac (FAILED - 1)

Failures:

  1) cf-uaac image has the expected version of uaac
     Failure/Error:
       expect(
         command("uaac --version").stdout.strip
       ).to eq("UAA client #{UAAC_VERSION}")
       
       expected: "UAA client 3.2.0"
            got: ""
       
       (compared using ==)
       
     # ./cf-uaac/cf-uaac_spec.rb:13:in `block (2 levels) in <top (required)>'

Finished in 0.70663 seconds (files took 1.54 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./cf-uaac/cf-uaac_spec.rb:12 # cf-uaac image has the expected version of uaac

```

**Comment by bandesz** [comment]
@henrytk everything passed for me, on Travis and on Docker Hub, so this might be a transient error in your build. Could you please include your build logs?

**Comment by bandesz** [comment]
@henrytk also you're running Ruby 2.2 locally: `/Users/henryknott/.rvm/rubies/ruby-2.2.1/bin/ruby`

**Comment by henrytk** [comment]
Caused by this: https://github.com/docker-library/ruby/pull/209

**Comment by bandesz** [comment]
@henrytk the failure is caused by the latest ruby:2.5-alpine Docker image update: https://github.com/docker-library/ruby/pull/209. I pushed a temporary fix for the cf-uaac image.

**Pull request merged**